Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -706,6 +398,311 @@ internal struct JSONScanner {
     return Character(UnicodeScalar(UInt32(currentByte))!)
   }
 
+  // Parse the leading UInt64 from the provided utf8 bytes.
+  //
+  // This usually does a direct conversion of utf8 to UInt64.  It is
+  // called for both unquoted numbers and for numbers stored in quoted
+  // strings.  In the latter case, the caller is responsible for
+  // consuming the leading quote and verifying the trailing quote.
+  //
+  // If the number is in floating-point format, this uses a slower
+  // and less accurate approach: it identifies a substring comprising
+  // a float, and then uses Double() and UInt64() to convert that
+  // string to an unsigned intger.
+  //
+  // If it encounters a "\" backslash character, it returns a nil.  This
+  // is used by callers that are parsing quoted numbers.  See nextSInt()
+  // and nextUInt() below.
+  private func parseBareUInt(
+    source: UnsafeBufferPointer<UInt8>,
+    index: inout UnsafeBufferPointer<UInt8>.Index,
+    end: UnsafeBufferPointer<UInt8>.Index
+  ) throws -> UInt64? {
+    let start = index
+    let c = source[index]
+    source.formIndex(after: &index)
```

Comment:
I think the code might get a little simpler if you skip this.  In the zero case before, advance it to check the next character, but in the double path, you reset it, so this was wasted.  Likewise, the case where it is a read number, you probably won't need need to start the number with the initial value, and can just use the loop direct seeding `digit` with zero.